runelite-client/src/main/java/net/runelite/client/plugins/worldmap/WorldMapPlugin.java

```diff
@@ -202,6 +236,40 @@ public void onXpChanged(ExperienceChanged event)
 		}
 	}
 
+	@Subscribe
+	public void onWidgetLoaded(WidgetLoaded event)
+	{
+		if (event.getGroupId() == QUEST_COMPLETED_GROUP_ID && config.questStartIcon())
+		{
+			worldMapPointManager.removeIf(QuestStartPoint.class::isInstance);
+			Arrays.stream(QuestStartLocation.values())
+				.map(value -> new QuestStartPoint(value, config.questStartIcon() && !value.isComplete() ? HIGHLIGHT_ICON : BLANK_ICON))
+				.forEach(worldMapPointManager::add);
+		}
+		else if (event.getGroupId() == WORLD_MAP_GROUP_ID && config.questStartIcon())
+		{
+			clientThread.invokeLater(() ->
```

Comment:
We are already in client thread?

Reply:
I’m not sure where to put this code.  Do you know where convention would put it?

Reply:
Yes, remove call to clientThread.invokeLater and just use the code

Reply:
It has to be on the client thread because it requests different varbits that aren't set yet when the client is launched

Reply:
So you are intentionally deferring it 1 client tick?

Reply:
You don't need to use `clientThread.invokeLater` to achieve this. Just use a flag and run your code in `onGameTick` when the flag is set.

Reply:
alright, will do

Reply:
invokeLater seems better than that tho